### PR TITLE
test: add superchain erc20 factory tests

### PR DESCRIPTION
**Pull request by agusduha**
Closes OPT-147
Closes OPT-148
Closes OPT-149
Closes OPT-150

### CODE REVIEW

File: packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol [hunk 1]
```diff
@@ -145,7 +142,7 @@ contract L2StandardBridgeInterop_LegacyToSuper_Test is L2StandardBridgeInterop_T
         _mockDeployments(address(l2OptimismMintableERC20Factory), _from, _remoteToken);
 
         // Mock the superchain factory to return address(0)
-        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _to, address(0));
+        _mockDeployments(address(l2OptimismSuperchainERC20Factory), _to, address(0));
```
Comment (0xParti) [inline packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol:145]: why did you remove the `Predeploys...` way of calling it?

File: packages/contracts-bedrock/test/L2/OptimismSuperchainERC20Factory.t.sol [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        vm.expectEmit(true, true, true, true);
+        emit OptimismSuperchainERC20Factory.OptimismSuperchainERC20Created(deployment, _remoteToken, _caller);
+
+        // Act
+        vm.prank(_caller);
+        address addr = superchainERC20Factory.deploy(_remoteToken, _name, _symbol, _decimals);
+
+        // Assert
+        assertTrue(addr == deployment);
+        assertTrue(OptimismSuperchainERC20(deployment).decimals() == _decimals);
```
Comment (0xParti) [inline packages/contracts-bedrock/test/L2/OptimismSuperchainERC20Factory.t.sol:76]: you can also assert for name symbol and remote token

**Review by 0xParti — APPROVED** [review]
Very nice! left small nits

File: packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol (hunk 1 above)
Comment (agusduha) [reply packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol:145]: I wanted to replicate the way they use the initialized predeploys across the different test suites

**Review by 0xParti — APPROVED** [review]
